blueman/plugins/applet/DBusService.py

```diff
@@ -78,8 +78,12 @@ def cb(_inst, ret):
     @dbus.service.method('org.blueman.Applet', in_signature="osd", async_callbacks=("ok", "err"))
     def disconnect_service(self, object_path, uuid, port, ok, err):
         if uuid == '00000000-0000-0000-0000-000000000000':
+            def on_timeout(dev):
+                dev.disconnect(reply_handler=ok, error_handler=err)
+
             device = Device(object_path)
-            device.disconnect(reply_handler=ok, error_handler=err)
+            self.parent.Plugins.run("on_device_disconnect", device)
```

Comment:
That does not seem right to me. Triggering `SerialManager.on_device_disconnect` here would mean that all connected serial services get disconnected when generic services (audio, HID) get disconnected.

I think what we actually need to do is transform `SerialManager.on_device_disconnect` into an `on_device_removed` method to kill the remaining watchers when the device got removed.

Reply:
Hmm, I'll look into it.

Reply:
Thought about it and there is no way we can keep the serial devices connected. Generic disconnect will actually disconnect the device and any services setup will be forcefully disconnected. 

I can however move this out of DBusService and have SerialManager handle when the device Connected property changes to False. It makes more sense to do this as the only user of ``on_device_disconnected`` is SerialManager.